src/Symfony/Component/Lock/Store/FlockStore.php

```diff
@@ -32,12 +32,15 @@ class FlockStore implements BlockingStoreInterface, SharedLockStoreInterface
 {
     private ?string $lockPath;
 
+    private ?string $prefix;
+
     /**
      * @param string|null $lockPath the directory to store the lock, defaults to the system's temporary directory
+     * @param string|null $prefix   a prefix to add to the lock filenames to avoid collision
      *
      * @throws LockStorageException If the lock directory doesn’t exist or is not writable
      */
-    public function __construct(?string $lockPath = null)
+    public function __construct(?string $lockPath = null, ?string $prefix = null)
```

Comment:
```suggestion
    public function __construct(
        ?string $lockPath = null, 
        private ?string $prefix = null,
    )
```

We can use CPP (constructor property promotion here)